### PR TITLE
Feature/icat backend generator#369

### LINKED ISSUE #369: Migrate Data Generator Script to use ICAT Backend
**Description**:
Within `util/`, there's a generator script which generates sample data to be used on SciGateway preprod. This script uses the API's database backend to insert data into the database. This backend isn't used in production and as time passes, will become more and more out of date, therefore being harder to maintain. As such, we should change the generator script to use the ICAT backend.

**Acceptance criteria**:
- [ ] Use the ICAT backend/Python ICAT to put sample data into the database
- [ ] The script should generate the same data as the original. The `icatdb Generator Script Consistency Test` CI job will test for this, although you could test manually by diffing API responses to see the data is the same

### PR DESCRIPTION
**Pull request by Reillyhewitson**
This PR will close #369 

## Description
This PR updates the generator script to use the python_icat backend, it also updates the dg-api and search-api tests so that the new data passes.

## Testing Instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check GitHub Actions build
- [ ] If `icatdb Generator Script Consistency Test` CI job fails, is this because of a deliberate change made to the script to change generated data (which isn't actually a problem) or is here an underlying issue with the changes made?
- [ ] Review changes to test coverage
- [ ] Does this change mean a new patch, minor or major version should be made? If so, does one of the commit messages feature `fix:`, `feat:` or `BREAKING CHANGE:` so a release is automatically made via GitHub Actions upon merge?
- [ ] {more steps here}

## Agile Board Tracking
Connect to #369

### CODE REVIEW

File: util/icat_db_generator.py [hunk 1]
```diff
@@ -610,41 +637,50 @@ class DatasetParameterGenerator(Generator):
...
         for i in range(1, self.amount):
             DatafileGenerator.generate_datafile(i)
-
-    @staticmethod
-    def generate_datafile(i):
-        datafile = models.DATAFILE()
-        apply_common_attributes(datafile, i % 19)
+        # timer = datetime.datetime.now()
+        # with multiprocessing.get_context("spawn").Pool() as pool:
+        #    pool.map(DatafileGenerator.generate_datafile, range(1, self.amount))
```
Comment (MRichards99) [inline util/icat_db_generator.py:662]: Can this commented out code be removed?

File: util/icat_db_generator.py [hunk 2]
```diff
@@ -653,78 +689,78 @@ class PermissibleStringValueGenerator(Generator):
...
-        apply_common_parameter_attributes(datafile_param, i)
-        datafile_param.datafileID = i
-        datafile_param.parameterTypeID = faker.random_int(
-            1, ParameterTypeGenerator.amount - 1,
+        # timer = datetime.datetime.now()
+        # with multiprocessing.get_context("spawn").Pool() as pool:
+        #    pool.map(
+        #        DatafileParameterGenerator.generate_datafile_parameter,
+        #        range(1, self.amount),
+        #    )
```
Comment (MRichards99) [inline util/icat_db_generator.py:750]: Can this commented out code be removed?

**Review by MRichards99 — CHANGES_REQUESTED** [review]
This looks great, a really good piece of work that I know wasn't the most interesting thing to do. I've made a couple of minor comments on the generator script but it's great that it works!

File: util/icat_db_generator.py [hunk 3]
```diff
@@ -84,29 +62,47 @@ def get_end_date(i):
...
-    entity.modTime = get_date_time()
-    entity.createTime = get_date_time()
-    entity.name = f"{entity.__tablename__} {iterator}"
-    entity.description = faker.text()
-    entity.facilityID = 1
-    entity.startDate = get_start_date(iterator)
-    entity.endDate = get_end_date(iterator)
-    entity.doi = faker.isbn10(separator="-")
+def validate():
+    pass
```
[the remark this answers is not in the document]
Comment (Reillyhewitson) [reply util/icat_db_generator.py]: It was meant to see if I could bypass the icat validation applied by python_icat. It was only being used by keywords and didn't significantly change the performance so I've removed the code.

File: util/icat_db_generator.py (hunk 2 above)
Comment (Reillyhewitson) [reply util/icat_db_generator.py]: Yes, removed.

File: util/icat_db_generator.py (hunk 1 above)
Comment (Reillyhewitson) [reply util/icat_db_generator.py:654]: Yes, removed.

**Review by MRichards99 — APPROVED** [review]

**Review by MRichards99 — APPROVED** [review]

**Pull request merged**